src/rabbit_vhost_dead_letter.erl

```diff
@@ -0,0 +1,135 @@
+%% The contents of this file are subject to the Mozilla Public License
+%% Version 1.1 (the "License"); you may not use this file except in
+%% compliance with the License. You may obtain a copy of the License
+%% at http://www.mozilla.org/MPL/
+%%
+%% Software distributed under the License is distributed on an "AS IS"
+%% basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See
+%% the License for the specific language governing rights and
+%% limitations under the License.
+%%
+%% The Original Code is RabbitMQ.
+%%
+%% The Initial Developer of the Original Code is GoPivotal, Inc.
+%% Copyright (c) 2018 Pivotal Software, Inc.  All rights reserved.
+%%
+
+-module(rabbit_vhost_dead_letter).
```

Comment:
I have a bad feeling about this. What could be a performance impact of having a single process handling all dead-letters from multiple queues?

Reply:
We'd have to test this. We had a team discussion about this change where we concluded that a per vhost dlx processes was likely to be fine but that further testing would be needed to confirm.